ibm/service/database/resource_ibm_database.go

```diff
@@ -1702,6 +1704,51 @@ func resourceIBMDatabaseInstanceCreate(context context.Context, d *schema.Resour
 		}
 	}
 
+	if _, ok := d.GetOk("logical_replication_slot"); ok {
+		service := d.Get("service").(string)
+		if service != "databases-for-postgresql" {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error Logical Replication can only be set for databases-for-postgresql instances"))
+		}
+
+		cloudDatabasesClient, err := meta.(conns.ClientSession).CloudDatabasesV5()
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error getting database client settings: %s", err))
+		}
+
+		_, logicalReplicationList := d.GetChange("logical_replication_slot")
+
+		if logicalReplicationList == nil {
```

Comment:
I don't think we need this nil check. We've already checked that `logical_replication_slot` is set.